### ISSUE TITLE
Standalone protocolCraft?

### ISSUE THREAD

**Issue by maxsupermanhd** [adepierre/Botcraft#29]
I would really like to have it as a module with separated repository.

**Comment by adepierre** [comment]
Could you please detail what you want to do and why you can't with the current library?

As an example you can check [Sniffcraft](https://github.com/adepierre/SniffCraft) which only uses the protocolCraft library from this repo.

**Comment by maxsupermanhd** [comment]
Currently protocolCraft is bound to this repo that contains a bunch of unwanted code and mess that I do not really want to deal with, at the same time just copying over protocolCraft directory kinda solves the issue however if I want to use it witin my projects I need to pull whole repository for that (ofc not conflicting with license). Having a dedicated repository will free me and everyone else from pulling updates of bot intefaces and other glues that you are adding (not against them, they are pretty good).

**Comment by adepierre** [comment]
Yeah I get your point, and I agree it's not ideal that you have to download the whole code just for protocolCraft. It's been on my TODO list for a while now.

On the other hand it's just some extra code, but you don't miss any feature so it's not really a _big_ problem. Isolating protocolCraft in its own repo would require to duplicate some cmake options too (like GAME_VERSION) and rework the dependencies system for picoJson as it's shared by both libraries.

So eventually this will be done, but it's not on my list of things to do in priority unfortunately.
